src/couch_mrview/src/couch_mrview_http.erl

```diff
@@ -380,22 +389,49 @@ view_cb({meta, Meta}, #vacc{meta_sent=false, row_sent=false}=Acc) ->
     {ok, AccOut} = maybe_flush_response(Acc, Chunk, iolist_size(Chunk)),
     {ok, AccOut#vacc{prepend="", meta_sent=true}};
 
-view_cb({meta, _Meta}, #vacc{}=Acc) ->
+streaming_cb({meta, _Meta}, #vacc{}=Acc) ->
     %% ignore metadata
     {ok, Acc};
 
-view_cb({row, Row}, #vacc{meta_sent=false}=Acc) ->
+streaming_cb({row, Row}, #vacc{meta_sent=false}=Acc) ->
     %% sorted=false and row arrived before meta
     % Adding another row
     Chunk = [prepend_val(Acc), "{\"rows\":[\r\n", row_to_json(Row)],
     maybe_flush_response(Acc#vacc{meta_sent=true, row_sent=true}, Chunk, iolist_size(Chunk));
 
-view_cb({row, Row}, #vacc{meta_sent=true}=Acc) ->
+streaming_cb({row, Row}, #vacc{meta_sent=true}=Acc) ->
     % Adding another row
     Chunk = [prepend_val(Acc), row_to_json(Row)],
     maybe_flush_response(Acc#vacc{row_sent=true}, Chunk, iolist_size(Chunk)).
 
 
+paginated_cb({row, Row}, #vacc{buffer=Buf}=Acc) ->
+    {ok, Acc#vacc{buffer = [row_to_obj(Row) | Buf]}};
+
+paginated_cb({error, Reason}, #vacc{}=_Acc) ->
+    throw({error, Reason});
+
+paginated_cb(complete, #vacc{buffer=Buf}=Acc) ->
+    {ok, Acc#vacc{buffer=lists:reverse(Buf)}};
+
+paginated_cb({meta, Meta}, #vacc{}=VAcc) ->
+    MetaMap = lists:foldl(fun(MetaData, Acc) ->
+        case MetaData of
+            {_Key, undefined} ->
+                Acc;
+            {total, _Value} ->
+                %% We set total_rows elsewere
+                Acc;
+            {Key, null} ->
+                maps:put(
+                    list_to_binary(atom_to_list(Key)), <<"null">>, Acc);
```

Comment:
You're converting a `null` to `"null"` here which seems wrong. That is to say, you've gone from the `null` terminal type, to a string with the letters null.